### PR TITLE
Change Displays Schedule

### PR DESCRIPTION
**Pull request by ezequielc**
## Description
This is the initial implementation of Change Schedule feature for Display details. 
The basic functionality is implemented but I will use this PR as base for other improvements.

## Motivation and Context
Display Improvements epic.

## How Has This Been Tested?
Locally and on [stage-1].

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
If any Release Checklist items were intentionally skipped, please provide which ones and the reasons why

### CODE REVIEW

**Comment by ezequielc** [comment]
@alex-deaconu I opened this PR to get an early feedback on the code. I still need to handle edge cases, errors, unit tests and styling. Will work on them in subsequent PRs. Let me know if you have any feedback. thanks!

File: web/scripts/displays/controllers/ctr-display-details.js [hunk 1]
```diff
@@ -19,25 +19,37 @@ angular.module('risevision.displays.controllers')
...
+      $scope.$watchGroup(['factory.loadingDisplay','scheduleFactory.savingSchedule'], function (loading) {
+        if (!$scope.factory.loadingDisplay && !$scope.scheduleFactory.savingSchedule) {
           $loading.stop('display-loader');
+        } else {
+          $loading.start('display-loader');
         }
       });
 
+      $scope.onScheduleChanged = function(newSchedule) {
+        $scope.selectedSchedule = newSchedule;
```
Comment (alex-deaconu) [inline web/scripts/displays/controllers/ctr-display-details.js:50]: @ezequielc curious why have a function when we can just update the `selectedSchedule` variable from the directive? Thanks.

File: web/scripts/schedules/directives/dtv-preview-selector.js [hunk 2]
```diff
@@ -1,15 +1,27 @@
...
-  .directive('previewSelector', ['$timeout', '$loading',
-    function ($timeout, $loading) {
+  .directive('previewSelector', ['$timeout', '$loading', 'ScrollingListService', 'schedule',
+    function ($timeout, $loading, ScrollingListService, schedule) {
       return {
-        restrict: 'A',
+        restrict: 'E',
+        templateUrl: 'partials/schedules/preview-selector.html',
+        scope: {
+          selectedSchedule: '=',
```
Comment (alex-deaconu) [inline web/scripts/schedules/directives/dtv-preview-selector.js:10]: @ezequielc as per the above, I would just update the variable with the selected schedule and not use a callback. Also, would we benefit from using `ng-model`? Thanks.

File: web/scripts/displays/controllers/ctr-display-details.js [hunk 3]
```diff
@@ -184,9 +196,31 @@ angular.module('risevision.displays.controllers')
...
-          var shouldSkipAddressValidation = !addressService.isAddressFormDirty($scope.displayDetails) && !$scope
-            .displayDetails.useCompanyAddress.$dirty;
-          return displayFactory.updateDisplay(shouldSkipAddressValidation);
+          var shouldSkipAddressValidation = !addressService.isAddressFormDirty($scope.displayDetails) && !$scope.displayDetails.useCompanyAddress.$dirty;
+          return displayFactory.updateDisplay(shouldSkipAddressValidation).then(_saveSchedule);
+        }
+      };
+
+      var _saveSchedule = function() {
+        var schedule = $scope.selectedSchedule;
```
Comment (alex-deaconu) [inline web/scripts/displays/controllers/ctr-display-details.js:205]: @ezequielc I would move this functionality to `scheduleFactory`, or maybe its own factory. Thanks.

File: web/scripts/schedules/services/svc-schedule-factory.js [hunk 4]
```diff
@@ -216,16 +221,20 @@ angular.module('risevision.schedules.services')
...
-            return _showDistributionConflictModal()
-              .then(function () {
-                return $q.resolve(schedule.update(_scheduleId, factory.schedule, true));
-              });
-          }
-          return $q.reject(err);
-        });
+      var _updateSchedule = function (forceDistribution) {
+        if (forceDistribution) {
+          return $q.resolve(schedule.update(_scheduleId, factory.schedule, true));
```
Comment (alex-deaconu) [inline web/scripts/schedules/services/svc-schedule-factory.js:226]: @ezequielc I would refactor this to its own function that can be called directly (maybe `factory.forceUpdateSchedule`). Thanks.

File: web/scripts/schedules/services/svc-schedule-factory.js [hunk 5]
```diff
@@ -42,6 +42,11 @@ angular.module('risevision.schedules.services')
         _init();
       };
 
+      factory.setSchedule = function (schedule) {
```
Comment (alex-deaconu) [inline web/scripts/schedules/services/svc-schedule-factory.js:45]: @ezequielc wondering if we could skip this step; either with a new factory, or passing the schedule to be updated to the `forceUpdateSchedule` function (see below) and if missing use the factory object? Thanks.

**Comment by alex-deaconu** [comment]
@ezequielc looks good, no major concerns, just some ideas for improvement. Thanks.

File: web/scripts/schedules/directives/dtv-preview-selector.js (hunk 2 above)
Comment (ezequielc) [reply web/scripts/schedules/directives/dtv-preview-selector.js:10]: @alex-deaconu Thanks, as per previous comment, I will try to improve this. Thanks

File: web/scripts/schedules/services/svc-schedule-factory.js (hunk 4 above)
Comment (ezequielc) [reply web/scripts/schedules/services/svc-schedule-factory.js:226]: @alex-deaconu Yeah, I thought about that too but I was trying to avoid replicating `updateSchedule()` code. Maybe I can extract a private method and use for both. Will try to improve it as well. Thanks!

File: web/scripts/schedules/services/svc-schedule-factory.js (hunk 5 above)
Comment (ezequielc) [reply web/scripts/schedules/services/svc-schedule-factory.js:45]: Good idea, will try that. Thanks

File: web/scripts/displays/controllers/ctr-display-details.js (hunk 3 above)
Comment (ezequielc) [reply web/scripts/displays/controllers/ctr-display-details.js:205]: Good point, I think I will move to `scheduleFactory`. Thanks

**Comment by ezequielc** [comment]
@alex-deaconu Great suggestions, I will work on them soon. Thanks

**Comment by ezequielc** [comment]
@alex-deaconu this is ready for a final review and merging. Thanks!

File: web/partials/displays/display-fields.html [hunk 6]
```diff
@@ -449,62 +449,37 @@
...
-            tooltip-digest-on-resize
-          >
-          	<div class="row-entry">
-              <span class="text-muted mr-auto" ng-if="!display.scheduleId">Create or select a schedule</span>
-              <span class="u_text-ellipsis mr-auto" ng-if="display.scheduleId">{{display.scheduleName}}</span>
-              <span class="text-muted pl-2"><i class="fa fa-sort"></i></span>
-            </div>
-          </button> -->
-        </div>
+  <!-- TODO: Remove madero-style once it's global for the page -->
```
Comment (alex-deaconu) [inline web/partials/displays/display-fields.html:452]: @ezequielc this has been corrected so the TODO can be removed. No need to push another commit/build to this branch just a note. Thanks.

File: web/partials/schedules/preview-selector.html [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<div class="form-group {{additionalClass}}">
```
Comment (alex-deaconu) [inline web/partials/schedules/preview-selector.html:1]: @ezequielc can't the `additionalClass` be applied to the `<preview-selector>` element with the same results (only margins from what I can tell). Thanks.

File: web/scripts/schedules/services/svc-schedule-factory.js [hunk 8]
```diff
@@ -189,6 +189,40 @@ angular.module('risevision.schedules.services')
         return deferred.promise;
       };
 
+      factory.forceUpdateSchedule = function (providedSchedule) {
```
Comment (alex-deaconu) [inline web/scripts/schedules/services/svc-schedule-factory.js:192]: @ezequielc looking at the way this turned out, maybe we can have a separate factory to handle force update & adding to distribution. Just a thought. Thanks.

**Comment by ezequielc** [comment]
@alex-deaconu Sounds good, I added the suggestions to our TODO list and we can work on them later. Thanks!
